### DISCUSSION TITLE
get the filename from NEW in storage trigger

### DISCUSSION THREAD

**Discussion by dongumayagay** [supabase/supabase#18286] [Questions] [answered]
i want to sync user uploaded files with table. but having hard time getting properties from NEW.
so i think the user_id is in NEW.owner, but now having trouble with filename as having error that it doesn't exist on NEW
i tried NEW.metadata->'filename' and 'name'

**Comment by GaryAustin1** [comment, marked as answer]
You can look in the table UI at the storage schema and objects table.  It will show the column names you get in NEW.
Note that last I checked storage-api does an insert of basic info to test RLS.  Then it writes the file to s3.  If that succeeds then it does an  update to the table to add file metadata.  If it fails then it deletes the row from the table.   So you could get an insert and then a delete and have no file stored.

**Reply by GaryAustin1** [reply]
You can also use `raise log 'new = %',new;` in your insert trigger after begin and then look in the Postgres log in the dashboard to see what  data is being inserted as some is on insert and some is on update.

**Reply by dongumayagay** [reply]
Thanks!
